autopilot/autopilot.go

```diff
@@ -463,11 +470,17 @@ func (ap *Autopilot) blockUntilSynced(interrupt <-chan time.Time) (synced, block
 }
 
 func (ap *Autopilot) tryScheduleTriggerWhenFunded() error {
-	ctx, cancel := context.WithTimeout(ap.shutdownCtx, 30*time.Second)
-	wallet, err := ap.bus.Wallet(ctx)
-	cancel()
+	// no need to schedule a trigger if we're stopped
+	if ap.isStopped() {
```

Comment:
Returning `nil` here on shutdown is a bit of an anti pattern since it doesn't prevent `ap.bus.Wallet` from returning an error due to a race.
Imo the better way would be to check `tryScheduleTriggerWhenFunded`'s return value for `context.Canceled`.

Reply:
The goroutine could also slightly change to not use a context derived from `ap.shutdownCtx` to make sure we don't log an error there.

Reply:
went the `context.Canceled` route 👍 don't think we need to add a `context.Background` here if we can avoid it